spec/unit/services/add_product_to_basket_spec.rb

```diff
@@ -0,0 +1,25 @@
+require_relative "../../../lib/services/add_product_to_basket"
+require_relative "../../../lib/model/basket"
+require_relative "../../support/shared_examples/models"
+
+RSpec.describe Store::AddProductToBasket do
+  include_examples :models
+
+  let(:quantity) { 2 }
+  let(:product_id) { 1 }
+  let(:new_basket) { Store::Basket.new(product_id: product_id, quantity: quantity) }
```

Comment:
those Basket model is not the best name, it's more Item in Basket then Basket itself. If you go to the shop you do not put basket in baskets. You put items in basket